Python/pystate.c

```diff
@@ -763,10 +763,10 @@ interpreter_clear(PyInterpreterState *interp, PyThreadState *tstate)
 
     Py_CLEAR(interp->audit_hooks);
 
-    // At this time, all the threads should be cleared so we don't need atomic
-    // operations for instrumentation_version or eval_breaker.
+    // Daemon threads may still access eval_breaker atomically via take_gil().
+    // Use atomic store to prevent data races during finalization.
```

Comment:
```suggestion
    // gh-140257: Threads have already been cleared, but daemon threads may
    // still access eval_breaker atomically via take_gil() right before they
    // hang. Use an atomic store to prevent data races during finalization.
```